### PR TITLE
update model for python3 and modeldb-ci support

### PR DESCRIPTION
**Pull request by kotsaloscv**
Actions followed (copied from this PR: https://github.com/neuronsimulator/nrn-modeldb-ci/pull/41):
    - 2to3 -w .
    - reindent .
    # TODO - PR to update model for python3 support
    - sed -i 's/figure, hold, isscalar/figure, isscalar/g;s/hold(True)/#hold(True)/g' analysis.py
    - sed -i 's/hashlib.md5(obj)/hashlib.md5(obj.encode("utf-8"))/g' shared.py
    - sed -i 's/cell.t0 = .0/#cell.t0 = .0/g' izhi.py
    - sed -i "s/(nCells\/s.nMuscles)/int(nCells\/s.nMuscles)/g;s/'w'/'wb'/g;s/s.hostspikevecs\[c\]/\[s.hostspikevecs\[c\].x\[i\] for i in range(s.hostspikevecs\[c\].size())\]/g" network.py

### CODE REVIEW

File: izhi.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def createcell(section, C, k, vr, vt, vpeak, a, b, c, d, celltype, cellid):
     cell.d = d
     cell.celltype = celltype # Set cell celltype (used for setting celltype-specific dynamics)
     cell.cellid = cellid # Cell ID for keeping track which cell this is
-    cell.t0 = .0
+    #cell.t0 = .0
```
Comment (alexsavulescu) [inline izhi.py:51]: forgot why this is commented out

File: main.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import matplotlib
-# matplotlib.use('Agg') # needed for hpc batch sims
+matplotlib.use('Agg') # needed for hpc batch sims
```
Comment (alexsavulescu) [inline main.py]: since `Agg` has no UI I think it should be kept as-is

File: izhi.py (hunk 1 above)
Comment (kotsaloscv) [reply izhi.py:51]: if I do not comment it out I get the following error:
```
python main.py

Setting parameters...
Benchmarking...
  Running at 114% default speed (114% total)

Creating simulation of 8512 cells for 1.0 s on 1 hosts...
Traceback (most recent call last):
  File "/gpfs/bbp.cscs.ch/data/project/proj16/kotsalos/HACKRON2022/nrn-modeldb-ci/test/Dura-BernalEtAl2017/main.py", line 40, in <module>
    network.runTrainTest2targets()
  File "/gpfs/bbp.cscs.ch/data/project/proj16/kotsalos/HACKRON2022/nrn-modeldb-ci/test/Dura-BernalEtAl2017/network.py", line 78, in runTrainTest2targets
    createNetwork()
  File "/gpfs/bbp.cscs.ch/data/project/proj16/kotsalos/HACKRON2022/nrn-modeldb-ci/test/Dura-BernalEtAl2017/network.py", line 333, in createNetwork
    cell = celltypes[gid](s.dummies[s.cellsperhost], cellid=gid) # Create a new cell of the appropriate type (celltypes[gid]) and store it
  File "/gpfs/bbp.cscs.ch/data/project/proj16/kotsalos/HACKRON2022/nrn-modeldb-ci/test/Dura-BernalEtAl2017/izhi.py", line 57, in RS
    cell = createcell(section, C, k, vr, vt, vpeak, a, b, c, d, celltype, cellid)
  File "/gpfs/bbp.cscs.ch/data/project/proj16/kotsalos/HACKRON2022/nrn-modeldb-ci/test/Dura-BernalEtAl2017/izhi.py", line 51, in createcell
    cell.t0 = .0
LookupError: 't0' is not a defined hoc variable name.
```

File: main.py (hunk 2 above)
Comment (kotsaloscv) [reply main.py]: fixed

File: izhi.py (hunk 1 above)
Comment (alexsavulescu) [reply izhi.py:51]: Ok will check on my side

Comment (alexsavulescu) [reply izhi.py:51]: So `t0` from `izhi2007.mod` is ASSIGNED only therefore it's used for internal calculations:

```bash
./x86_64/special
NEURON -- VERSION 8.2.0+ HEAD (156b9dee+) 2022-07-01
Duke, Yale, and the BlueBrain Project -- Copyright 1984-2021
See http://neuron.yale.edu/neuron/credits

Additional mechanisms from files
 "izhi2007.mod" "nsloc.mod" "stdp.mod" "vecevent.mod"
oc>objref c
oc>c = new Izhi2007()
oc>c.t0 = 1
t0 not a public member of Izhi2007
./x86_64/special: Izhi2007 t0
 near line 4
 c.t0 = 1
         ^
oc>
```
Most probably older NEURON versions allowed accessing ASSIGNED variables. So there are two solutions:

- remove `cell.t0 = 0`
- add `t0` to the `NEURON` block in `izhi2007.mod` (I'm leaning more towards this solution)

However not sure of the implications wrt the model itself. @ramcdougal @nrnhines any idea?

**Comment by nrnhines** [comment]
There is no downside to declaring RANGE t0 in the mod file.

File: server.py [hunk 3]
```diff
@@ -413,46 +413,46 @@ def callback(self):
...
         finally: # update current neuron time
             if isCommunication == 0:
                 pass
             else:
                 if newCurrTime.value + nextInvl <= duration:
-                    h.cvode.event(newCurrTime.value + nextInvl, self.callback)    
+                    h.cvode.event(newCurrTime.value + nextInvl, self.callback)
                 if verbose:
-                    print('[callback.h.t End(id= %i)] %.3f' %(pc.id(), h.t))                   
+                    print(('[callback.h.t End(id= %i)] %.3f' %(pc.id(), h.t)))
```
Comment (ramcdougal) [inline server.py:453]: pedantic point, but this change isn't necessary; it was already in parentheses

Comment (alexsavulescu) [reply server.py:453]: I think it's a side effect from the `2to3` utility.

**Pull request merged**